lib/providers/env_var_provider.ex

```diff
@@ -58,27 +58,40 @@ defmodule EnvVar.Provider do
   in the config files for values that are configured in the EnvVar
   Provider.
   """
-  def init(prefix: prefix, env_map: env_map, enforce: enforce) when is_atom(prefix) do
-    process_config(env_map, prefix, enforce)
-  end
+  def init(opts) do
+    env_map =
+      case Keyword.fetch!(opts, :env_map) do
+        map when is_map(map) -> map
+        other -> raise ArgumentError, ":env_map should be a map, got: #{inspect(other)}"
+      end
+
+    prefix =
+      case Keyword.fetch!(opts, :prefix) do
+        atom when is_atom(atom) -> atom
+        binary when is_binary(binary) -> String.to_atom(binary)
+        other -> raise ArgumentError, ":prefix should be atom or string, got: #{inspect(other)}"
+      end
 
-  def init(prefix: prefix, env_map: env_map, enforce: enforce) do
-    prefix = String.to_atom(prefix)
-    process_config(env_map, prefix, enforce)
+    enforce? =
+      case Keyword.get(opts, :enforce, true) do
+        bool when is_boolean(bool) -> bool
+        other -> raise ArgumentError, ":enforce should be a boolean, got: #{inspect(other)}"
+      end
+
+    process_config(env_map, prefix, enforce?)
   end
 
-  def show_vars(prefix: prefix, env_map: env_map) when is_atom(prefix) do
+  def show_vars(opts) do
+    prefix = opts |> Keyword.fetch!(:prefix) |> String.to_atom()
+    env_map = Keyword.fetch!(opts, :env_map)
+
```

Comment:
This doesn't enforce that both args are present and doesn't return any errors in that case. The previous implementation did.

Reply:
Wait I'm confused. This enforces that both the `:prefix` and `:env_map` options are there. Are referring to something else as well? 🤔

Reply:
Sorry, forgot that `!` means fetch raises.